### PR TITLE
ci: create a job to validate the PR  title

### PR DESCRIPTION
**Pull request by pedromtec**
## What's the purpose of this pull request?

Create a workflow with a job to validate PR title. The title should follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification. 

Available types:
```
 - fix
 - feat
 - chore
 - docs
 - style
 - refactor
 - test
 - ci
```

## How it works?
Captura de Tela 2024-12-03 às 14 35 33

## How to test it?

Change the title of this PR and refresh the page. When I update the title, I need to refresh the page to see the job running again. It looks like a visual issue.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, edited]
+
+jobs:
+  main:
+    name: Validate PR title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v5
```
Comment (eduardoformiga) [inline .github/workflows/pr.yml:12]: Is it trustworthy?

Comment (eduardoformiga) [reply .github/workflows/pr.yml:12]: can we add something like just to double-check the run only title changes:
`if: ${{ github.event.changes.title.from && github.event.changes.title.from != github.event.pull_request.title }}`

**Review by eduardoformiga — APPROVED** [review]
LGTM!

I added some non-blocking questions along with the review.

**Comment by eduardoformiga** [comment]
Hey, I was thinking about the Major pattern from conventional commits. They suggest adding a `!` after the type (e.g., feat!: blablabla) or adding a label (BREAKING CHANGE) in the commit footer, but I think this pattern won't validate the PR title here (?)

**Review by gvc — CHANGES_REQUESTED** [review]
Can you add the configuration to limit the scopes that would be applicable to the conventional commits? 

I think:

```
with:
  scopes: |
    core
    ui
    cli
    api
    # .. and the rest of packages
```

at the end should do the trick. See the docs here: https://github.com/amannn/action-semantic-pull-request?tab=readme-ov-file#configuration

**Comment by pedromtec** [comment]
> Can you add the configuration to limit the scopes that would be applicable to the conventional commits?
> 
> I think:
> 
> ```
> with:
>   scopes: |
>     core
>     ui
>     cli
>     api
>     # .. and the rest of packages
> ```
> 
> at the end should do the trick. See the docs here: https://github.com/amannn/action-semantic-pull-request?tab=readme-ov-file#configuration

I was thinking about this @gvc, we need to define which prefixes we want to have in Faststore. Which can even be all prefixes. I would add one more point: Which prefixes generate versions? Because not all prefixes generate and I think we need documentation for that. I'm going to do this test and I'm going to start a thread with the team. cc @eduardoformiga

File: .github/workflows/pr.yml (hunk 1 above)
Comment (pedromtec) [reply .github/workflows/pr.yml:12]: in the context of GitHub Actions, the block types: [opened, reopened, synchronize, edited] specifies which pull_request events will trigger the workflow. Here's what each type means:

opened: The workflow is triggered when a new Pull Request is created.

reopened: The workflow runs when a previously closed Pull Request is reopened.

synchronize: The workflow runs when new commits are pushed to the branch associated with the Pull Request, such as when you add changes after opening the PR.

edited: The workflow is triggered when the Pull Request is edited. This includes changes to the title, description, or metadata.

We listening all theses events @eduardoformiga . So I think that's enough

Comment (lariciamota) [reply .github/workflows/pr.yml:12]: question: wouldn't it be enough to use `opened`, `reopened` and `edited`?

Comment (hellofanny) [reply .github/workflows/pr.yml:12]: It seems like `synchronize` isn't relevant to our case 🤔

Comment (pedromtec) [reply .github/workflows/pr.yml:12]: It makes sense. I just follow the action documentation. But I can't see a case where we can change the PR title through commits.

**Comment by pedromtec** [comment]
> Can you add the configuration to limit the scopes that would be applicable to the conventional commits?
> 
> I think:
> 
> ```
> with:
>   scopes: |
>     core
>     ui
>     cli
>     api
>     # .. and the rest of packages
> ```
> 
> at the end should do the trick. See the docs here: https://github.com/amannn/action-semantic-pull-request?tab=readme-ov-file#configuration

Guila, I had misunderstood haha. The scopes make a lot of sense, yes

File: .github/workflows/pr.yml (hunk 1 above)
Comment (pedromtec) [reply .github/workflows/pr.yml:23]: I found a case just now. If I change the action code I want it to run.

**Review by gvc — APPROVED** [review]

**Pull request merged**